### PR TITLE
Packaging pi code

### PR DESCRIPTION
**Pull request by gunturiCM**
## Main Changes

Packaging Python code running on Pi. Below is the changes list:
1. Moved folder `raspberry/estimators` to `raspberry/bzzz/estimators`.
2. Moved folder `raspberry/read_sbus` to `raspberry/bzzz/read_sbus`.
3. Moved folder `raspberry/simulations` to `raspberry/bzzz/simulations`.
4. Some minor code changes.

## Associated Issues
None

## Tests
Just repacking existing code.

### CODE REVIEW

File: raspberry/bzzz/read_sbus/radioDataParser.py [hunk 1]
```diff
@@ -57,21 +57,26 @@ def __init__(self):
...
+    def __radio_calib(self):
+        """Takes 
+        """
+        pass
+
+    def __map_radio_to_angle(self, x, stick_min=RADIO_STICK_MIN, stick_max=RADIO_STICK_MAX):
         """Maps radio stick positions to corresponding angles in radians linearly.
 
         :param x: Stick position data.
         :return: angle in radians.
```
Comment (alphaville) [inline raspberry/bzzz/read_sbus/radioDataParser.py:69]: Add `stick_min` and `stick_max` to the docs

File: raspberry/setup.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        long_description = '\n' + f.read()
+except FileNotFoundError:
+    long_description = DESCRIPTION
+
+setup(name=NAME,
+      version=VERSION,
+      description=DESCRIPTION,
+      long_description=long_description,
+      long_description_content_type='text/markdown',
+      author=['CM', 'Pantelis Sopasakis'],
```
Comment (alphaville) [inline raspberry/setup.py:31]: Why don't you add your name and Jamie's name?

File: raspberry/setup.py
Comment (alphaville) [inline raspberry/setup.py:1]: Does this work? Does it build properly when you do `pip install .` from within a virtual environment? And does the quadcopter work with the packaged code?

File: raspberry/bzzz/read_sbus/radioDataParser.py (hunk 1 above)
Comment (gunturiCM) [reply raspberry/bzzz/read_sbus/radioDataParser.py:69]: It is documented in line 6.

File: raspberry/setup.py (hunk 2 above)
Comment (gunturiCM) [reply raspberry/setup.py:31]: Done!

File: raspberry/bzzz/read_sbus/radioDataParser.py (hunk 1 above)
Comment (alphaville) [reply raspberry/bzzz/read_sbus/radioDataParser.py:69]: This is the documentation I see for this function:

```
"""Maps radio stick positions to corresponding angles in radians linearly.
        :param x: Stick position data.
        :return: angle in radians.
"""
```

Strangely, there are some more docstring-like comments in lines 74-78.

File: raspberry/Kalman_filter_test.py
Comment (alphaville) [inline raspberry/Kalman_filter_test.py:1]: I don't understand what this stray file is. It is placed in `raspberry/`. Should it be part of the package? Is it a test?

Comment (gunturiCM) [reply raspberry/Kalman_filter_test.py:1]: Professor, `bzzz/` is the package. All the other files outside `bzzz/` and inside `raspberry/` are implementations.

Comment (alphaville) [reply raspberry/Kalman_filter_test.py:1]: Still, what is [`raspberry/Kalman_filter_test.py`](https://github.com/QUB-ASL/bzzz/pull/100/files/e2af43f1cdd21d968127ae6886389aced965ec9b#diff-7ca76327da5c6d7af91d741525f411a8f8dfb8dd591aecd76f9cd380273463cc)? Is it a test? If so, why isn't it in a test folder? Also tests should be excluded from the package in `setup.py`.

File: raspberry/bzzz/read_sbus/radioDataParser.py (hunk 1 above)
Comment (gunturiCM) [reply raspberry/bzzz/read_sbus/radioDataParser.py:69]: I see. Sorry, I was looking at a different line. I am on it.

File: raspberry/Kalman_filter_test.py
Comment (gunturiCM) [reply raspberry/Kalman_filter_test.py:1]: Oh! I understand. On it.

File: raspberry/setup.py
Comment (jamie-54) [reply raspberry/setup.py:1]: I checked that it builds properly on raspberry pi

File: raspberry/setup.py (hunk 2 above)
Comment (alphaville) [reply raspberry/setup.py:31]: It hasn't been done

File: raspberry/setup.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+version_file = open(os.path.join(here, 'VERSION'))
+VERSION = version_file.read().strip()
+
+DESCRIPTION = 'Bzzz is a quadcopter'
+
+
+# Import the README and use it as the long-description.
+# Note: this will only work if 'README.md' is present in your MANIFEST.in file!
+try:
+    with io.open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
```
Comment (alphaville) [inline raspberry/setup.py:21]: A `README.md` file is missing

File: raspberry/setup.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+from setuptools import setup, find_packages
+import io
+import os
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+NAME = 'bzzz'
+
+# Import version from file
+version_file = open(os.path.join(here, 'VERSION'))
```
Comment (alphaville) [inline raspberry/setup.py:12]: A `VERSION` file is missing

File: raspberry/bzzz.egg-info/PKG-INFO
Comment (alphaville) [inline raspberry/bzzz.egg-info/PKG-INFO:1]: remove egg.info

Comment (gunturiCM) [reply raspberry/bzzz.egg-info/PKG-INFO:1]: Done!

File: raspberry/setup.py (hunk 4 above)
Comment (gunturiCM) [reply raspberry/setup.py:12]: Done!

File: raspberry/setup.py (hunk 2 above)
Comment (gunturiCM) [reply raspberry/setup.py]: Done!

File: raspberry/Kalman_filter_test.py
Comment (gunturiCM) [reply raspberry/Kalman_filter_test.py:1]: Moved `raspberry/Kalman_filter_test.py` to `raspberry/bzzz/tests/Kalman_filter_test.py` and excluded from installation.

**Comment by alphaville** [comment]
I'm not sure when/whether the [altitude KF](https://github.com/QUB-ASL/bzzz/blob/feature/packaging_pi_code/raspberry/bzzz/estimators/altitude_Kalman_filter.py) was reviewed, but here are a few comments from my side:

- [ ] the constructor returns an object of type `bzzz.estimators.KalmanFilter`, not `None`
- [ ] the constructor requires documentation
- [ ] I would rename `run` to `update`
- [ ] In `run` rename `Tt` to something more understandable (`thrust_ref`?)
- [ ] change `10**-3` to `1e-3` (same for `10**-5`)
- [ ] `self.__cache_MU_values`: why is this a lambda?
- [ ] `MU_cache` - rename to `measurement_update_cache`
- [ ] how does this implementation of the KF handle missing values?

**Review by alphaville — APPROVED** [review]

**Pull request merged**